src/components/ProductBlock.tsx

```diff
@@ -143,129 +144,94 @@ class ProductBlock extends React.Component<IProps, IState> {
                 );
             });
         } else {
-            this.setState({ initial: false });
+            setInitial(false);
         }
     };
 
-    renderButtons = (initial: boolean, productBlock: iProductBlock) => {
-        const invalid = !initial && (this.isInvalid() || this.state.processing);
+    const renderButtons = (initial: boolean, productBlock: iProductBlock) => {
+        const invalid = !initial && (isInvalid() || processing);
         return (
-            <section className="buttons">
-                <EuiButton className="button" onClick={this.cancel}>
-                    <FormattedMessage id="metadata.productBlocks.back" />
-                </EuiButton>
-                {this.context.allowed(
-                    "/orchestrator/metadata/product-block/edit/" + productBlock.product_block_id + "/"
-                ) && (
-                    <EuiButton
-                        tabIndex={0}
-                        className={`button ${invalid ? "grey disabled" : "blue"}`}
-                        onClick={this.submit}
-                    >
-                        <FormattedMessage id="metadata.productBlocks.submit" />
+            <EuiFlexGroup className="buttons">
+                <EuiFlexItem grow={false}>
+                    <EuiButton className="button" onClick={cancel}>
+                        <FormattedMessage id="metadata.productBlocks.back" />
                     </EuiButton>
+                </EuiFlexItem>
+                {allowed("/orchestrator/metadata/product-block/edit/" + productBlock.product_block_id + "/") && (
```

Comment:
The `allowed(..)` result is stored (and updated) in `readOnly` so we could/should use that here, right? (and on the same check 10 lines below here)